fix pod availability checking logic

Hsn723 opened the pull request:
The changes to `IsPodAvailable` introduced in https://github.com/gruntwork-io/terratest/pull/480 incorrectly reported pods as available when the readinessProbe is failing but the containers were in a started state. Namely, when there is no startupProbe, `containerStatus.Started` will always be true ([ref](https://godoc.org/k8s.io/api/core/v1#ContainerStatus)) and the pod will be incorrectly reported as available.

This PR fixes the availability checking logic and adds a few test cases to validate the fix.

brikis98 commented on modules/k8s/pod.go:
```diff
@@ -142,7 +142,7 @@ func IsPodAvailable(pod *corev1.Pod) bool {
 		isContainerReady := containerStatus.Ready
 
-		if !isContainerReady && isContainerStarted != nil && !*isContainerStarted {
+		if !isContainerReady || isContainerStarted == nil || !*isContainerStarted {
```
NIT: `*isContainerStarted == false` is a tiny bit easier to read than `!*isContainerStarted`.

brikis98 reviewed (commented):
Thanks for adding the test cases!

This fix seems correct to me, but @yorinasub17, I'd appreciate a second set of eyes.

infraredgirl reviewed (dismissed):
LGTM!

yorinasub17 reviewed (dismissed):
This LGTM as well!

brikis98 commented:
Thank you all for the second (and third) sets of eyes!

@Hsn723 Could you fix the one NIT I mentioned, and pull in the latest from `master`? I can kick off tests then.

Hsn723 commented:
@brikis98 Thank you for the reviews. I have applied the requested fixes and synced to `master`

brikis98 reviewed (dismissed):
Great, thanks! I'll kick off tests now.

brikis98 commented:
Tests failed due to precommit hooks. Could you run `go fmt` on the code?

```
[INFO] Initializing environment for https://github.com/gruntwork-io/pre-commit.
goimports................................................................Failed
- hook id: goimports
- files were modified by this hook

modules/k8s/pod_test.go
```

Hsn723 commented:
@brikis98 You're right, I forgot to run gofmt. I've gofmt'ed the code now.

brikis98 commented:
Re-running tests!

brikis98 commented:
Hm, there were a number of test failures that seem to be related to waiting on pods...

Could you look into:

* `TestWaitUntilPodAvailableReturnsSuccessfully`
* `TestWaitUntilPodWithMultipleContainersAvailableReturnsSuccessfully`
* `TestWaitUntilPodAvailableWithReadinessProbe`
* `TestTunnelOpensAPortForwardTunnelToPod`
* `TestTunnelOpensAPortForwardTunnelToService`
* `TestRemoteChartInstall`
* `TestRemoteChartInstallUpgradeRollback`

Hsn723 commented:
@brikis98 Sorry, it looks like I misunderstood the documentation. Under Kubernetes v1.15 which is used in your CI, 
 `pod.Status.ContainerStatus.Started` is always `nil`. The reason for that is likely that startupProbes haven't been introduced yet (the documentation only showed the latest state of the Pod API...).

I've fixed it now (plus a typo with a test case I introduced), please take a look.

brikis98 reviewed (approved):
LGTM! I'll kick off tests again.

brikis98 commented:
Oh, actually, could you pull the latest from `master` first?

Hsn723 commented:
Sure thing. I've synced with `master`

brikis98 commented:
Thanks! I'll kick off tests shortly.

brikis98 reviewed (approved):
Two tests failed, but it seems to be some kind of transient issue with GCP compute instances, so I'm going to merge. Thanks!

brikis98 commented:
https://github.com/gruntwork-io/terratest/releases/tag/v0.30.27

Hsn723 commented:
Thank you!
